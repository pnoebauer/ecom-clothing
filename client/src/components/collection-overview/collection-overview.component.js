//loops over all product categories and maps CollectionPreview of each category
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { CollectionOverviewContainer } from './collection-overview.styles';

import CollectionPreview from '../collection-preview/collection-preview.component';
import { selectCollectionsOverview } from '../../redux/shop/shop.selectors';

const CollectionOverview = ({ collections }) => (
	<CollectionOverviewContainer>
		{collections.map(({ id, ...otherCollectionProps }) => (
			<CollectionPreview key={id} {...otherCollectionProps}/>
		))}
	</CollectionOverviewContainer>
); 

const mapStateToProps = createStructuredSelector({
	collections: selectCollectionsOverview
});

export default connect(mapStateToProps)(CollectionOverview)