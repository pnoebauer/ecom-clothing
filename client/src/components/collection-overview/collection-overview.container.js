import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import CollectionOverview from './collection-overview.component';
import WithSpinner from '../with-spinner/with-spinner.component';

import { selectIsCollectionLoaded } from '../../redux/shop/shop.selectors';


const mapStateToProps = createStructuredSelector({
	isLoading: state => !selectIsCollectionLoaded(state)
});

// const CollectionOverviewContainer = connect(mapStateToProps)(WithSpinner(CollectionOverview));
const CollectionOverviewContainer = compose(
	connect(mapStateToProps),
	WithSpinner
)(CollectionOverview);

export default CollectionOverviewContainer;