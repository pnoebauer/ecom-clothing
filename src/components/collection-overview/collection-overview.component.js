import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CollectionPreview from '../collection-preview/collection-preview.component';
import { selectCollections } from '../../redux/shop/shop.selectors';

import './collection-overview.styles.scss';

const CollectionOverview = ({ collections }) => (
	<div className='collection-overview'>
		{Object.entries(collections).map(([key, { id, ...otherCollectionProps }]) => (
			<CollectionPreview key={id} {...otherCollectionProps}/>
		))}
	</div>
); 

const mapStateToProps = createStructuredSelector({
	collections: selectCollections
});

export default connect(mapStateToProps)(CollectionOverview);

// {collections.map(({ id, ...otherCollectionProps }) => (
// 			<CollectionPreview key={id} {...otherCollectionProps}/>
// 		))}