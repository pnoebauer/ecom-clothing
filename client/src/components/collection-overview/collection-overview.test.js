import React from 'react';
import {shallow} from 'enzyme';
import {CollectionOverview} from './collection-overview.component';

const mockCollections = [
	{
		routeName: 'hats',
		id: '5XnhLO3MJTywoST6lftW',
		title: 'Hats',
		items: [
			{
				id: 1,
				imageUrl: 'https://i.ibb.co/ZYW3VTp/brown-brim.png',
				price: 25,
				name: 'Brown Brim',
			},
			{
				imageUrl: 'https://i.ibb.co/ypkgK0X/blue-beanie.png',
				id: 2,
				name: 'Blue Beanie',
				price: 18,
			},
		],
	},
	{
		routeName: 'sneakers',
		id: '6Ezkl920BOYZl5rzZ3ea',
		title: 'Sneakers',
		items: [
			{
				name: 'Adidas NMD',
				price: 220,
				id: 10,
				imageUrl: 'https://i.ibb.co/0s3pdnc/adidas-nmd.png',
			},
		],
	},
];

test('should render CollectionOverview component', () => {
	expect(shallow(<CollectionOverview collections={mockCollections} />)).toMatchSnapshot();
});
