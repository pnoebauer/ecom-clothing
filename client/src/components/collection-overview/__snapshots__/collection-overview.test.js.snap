// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CollectionOverview component 1`] = `
<styled.div>
  <CollectionPreview
    items={
      Array [
        Object {
          "id": 1,
          "imageUrl": "https://i.ibb.co/ZYW3VTp/brown-brim.png",
          "name": "Brown Brim",
          "price": 25,
        },
        Object {
          "id": 2,
          "imageUrl": "https://i.ibb.co/ypkgK0X/blue-beanie.png",
          "name": "Blue Beanie",
          "price": 18,
        },
      ]
    }
    key="5XnhLO3MJTywoST6lftW"
    routeName="hats"
    title="Hats"
  />
  <CollectionPreview
    items={
      Array [
        Object {
          "id": 10,
          "imageUrl": "https://i.ibb.co/0s3pdnc/adidas-nmd.png",
          "name": "Adidas NMD",
          "price": 220,
        },
      ]
    }
    key="6Ezkl920BOYZl5rzZ3ea"
    routeName="sneakers"
    title="Sneakers"
  />
</styled.div>
`;
